packages/safe-tools-client/app/components/schedule-payment-form-action-card/index.gts

```diff
@@ -29,16 +33,52 @@ export default class SchedulePaymentFormActionCard extends Component<Signature>
   }
   @tracked selectedPaymentType: string | undefined;
   @action onSelectPaymentType(paymentTypeId: string) {
+    if (paymentTypeId === 'one-time' && !this.paymentDate) {
+      this.paymentDate = new Date();
+    }
+    if (paymentTypeId === 'monthly') {
+      if (!this.monthlyUntil) {
+        let now = new Date();
+        this.monthlyUntil = new Date(now.getFullYear() + 1, now.getMonth(), now.getDate());
+      }
+      if (!this.paymentDayOfMonth) {
+        this.paymentDayOfMonth = 1;
+      }
+    }
     this.selectedPaymentType = paymentTypeId;
   }
 
+  @tracked paymentDate: Date | undefined;
+  @action onSetPaymentDate(day: Day) {
+    this.paymentDate?.setFullYear(day.getFullYear(), day.getMonth(), day.getDate()); 
+    this.paymentDate = this.paymentDate; // trigger reactivity
+  }
```

Comment:
There’s something strange happening for me where the highlighted value is one click behind:

![calendar-lag](https://user-images.githubusercontent.com/43280/201158003-6701bfef-afd7-4879-9428-bb275e234eab.gif)

It’s not happening [here](https://boxel.stack.cards/main/#/docs?s=Components&ss=%3CBoxel%3A%3AInput%3A%3ADate%3E) 😯

Reply:
Fixed.